src/components/app.js

```diff
@@ -105,6 +107,17 @@ class App extends React.Component {
             path="/createUser"
             render={() => (isValidUser ? <Redirect to="/editor" /> : <CreateUserPage />)}
           />
+          {/* Default error page */}
+          <Route
+            path="/error"
+            render={() =>
+              this.props.errorMsg ? (
```

Comment:
Bc of the error created above in onAuthHandler, if you're not logged in and go to /error and refresh, you'll remain on /error

Reply:
Should I just resolve this by deleting the path/error entirely? Am now unsure why in onAuthHandler, loadUserData is called with passing in showErrorPage.

Reply:
loadUserData is passed showErrorPage cuse the 2nd param is a function to be called if an error occurs

AppContainer.js
```
 loadUserData: async (uid, onFailure) => {
      const { ok, data, error } = await fetch.getUserData(uid, true);
      //if the request went fine, and there's a non empty userData
      if (ok && data && data.userData && Object.keys(data.userData).length) {
        dispatch(loadUserData(uid, data.userData));
        if (data.programs) {
          dispatch(loadPrograms(data.programs));
        }
      } else {
        onFailure(error);
      }
    },
```